x86: add 6.6 as testing kernel

namiltd opened the pull request:
Kernel 6.6 as the testing kernel for x86.

ehem committed:
kernel/x86: remove legacy 5.15 kernel config

Missed when removing the other 5.15 kernel configurations.

Fixes: b8ec063b38 ("x86: remove unused 5.15 kernel configs and patches")
Signed-off-by: Elliott Mitchell <ehem+openwrt@m5p.com>

namiltd commented:
Strange compilation errors:
```
Run make target/compile -j$(nproc) BUILD_LOG=1 || ret=$? .github/workflows/scripts/show_build_failures.sh
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a build dependency on 'libpam', which does not exist
WARNING: Makefile 'package/boot/kexec-tools/Makefile' has a dependency on 'liblzma', which does not exist
WARNING: Makefile 'package/network/services/lldpd/Makefile' has a dependency on 'libnetsnmp', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a build dependency on 'libpam', which does not exist
make[2]: Entering directory '/__w/openwrt/openwrt/openwrt/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/__w/openwrt/openwrt/openwrt/scripts/config'
 make[1] target/compile
 make[2] -C target/linux compile
    ERROR: target/linux failed to build.
make -r target/compile: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:233: target/compile] Error 1
====== Make errors from logs/target/linux/compile.txt ======
ld: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/check.o: in function `arch_is_rethunk':
/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/check.c:1299: multiple definition of `arch_is_rethunk'; /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/arch/x86/objtool-in.o:/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/arch/x86/decode.c:827: first defined here
ld: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/check.o: in function `arch_is_embedded_insn':
/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/check.c:1299: multiple definition of `arch_is_embedded_insn'; /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/arch/x86/objtool-in.o:/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/arch/x86/decode.c:832: first defined here
ld: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/special.o: in function `arch_handle_alternative':
/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/special.c:60: multiple definition of `arch_handle_alternative'; /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/arch/x86/objtool-in.o:/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/arch/x86/special.c:12: first defined here
ld: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/orc_gen.o: in function `orc_create':
/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/orc_gen.c:161: multiple definition of `orc_create'; /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/weak.o:/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/weak.c:22: first defined here
ld: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/orc_dump.o: in function `orc_dump':
/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/orc_dump.c:69: multiple definition of `orc_dump'; /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/weak.o:/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/weak.c:20: first defined here
make[8]: *** [/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/build/Makefile.build:156: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/objtool-in.o] Error 1
make[7]: *** [Makefile:67: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/tools/objtool/objtool-in.o] Error 2
make[6]: *** [Makefile:73: objtool] Error 2
make[5]: *** [/__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/Makefile:1355: tools/objtool] Error 2
make[4]: *** [Makefile:234: __sub-make] Error 2
make[3]: *** [Makefile:26: /__w/openwrt/openwrt/openwrt/build_dir/target-x86_64-openwrt-linux-musl_musl/linux-x86_64/linux-6.6.20/.modules] Error 2
make[2]: *** [Makefile:11: compile] Error 2
time: target/linux/compile#30.60#13.66#31.40
Error: Process completed with exit code 2.

namiltd committed:
x86: 6.6: refresh patches

Refresh kernel patches.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
x86: 6.6: set testing kernel

Allow selecting 6.6 as testing kernel on x86.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

vincejv commented:
Next error is solved by https://github.com/openwrt/openwrt/pull/14751#issuecomment-1986803975, as mentioned on previous comment

> Try adding +LINUX_6_6:kmod-crypto-rng to KernelPackage/crypto-manager depends as kmod-crypto-user is selecting it and it has a dependency on kmod-crypto-rng
> 
> For me on qualcommax all in-kernel kmods build and package fine so I cannot reproduce it easily.

trippleflux commented:
Somehow getting "Failed to build tmp/.config-package.in" when invoking "make dirclean".

vincejv commented:
Rebase this PR when this is merged #14885 should fix these issues

vincejv commented:
Also my suggestion to add `kmod-crypto-rng` to `kmod-crypto-user` was the correct fix, not the `kmod-crypto-rng` to `KernelPackage/crypto-manager` (what was quoted before), this is already applied on https://github.com/openwrt/openwrt/pull/14885

catsimple commented:
I merged https://github.com/openwrt/openwrt/pull/14885 locally and it didn't fix

```
catsimple@catsimple-virtual-machine:~/testingx86/openwrt$ make menuconfig
Checking 'true'... ok.
Checking 'false'... ok.
Checking 'working-make'... ok.
Checking 'case-sensitive-fs'... ok.
Checking 'proper-umask'... ok.
Checking 'gcc'... updated.
Checking 'working-gcc'... ok.
Checking 'g++'... updated.
Checking 'working-g++'... ok.
Checking 'ncurses.h'... ok.
Checking 'git'... updated.
Checking 'rsync'... updated.
Checking 'perl-data-dumper'... ok.
Checking 'perl-findbin'... ok.
Checking 'perl-file-copy'... ok.
Checking 'perl-file-compare'... ok.
Checking 'perl-thread-queue'... ok.
Checking 'perl-ipc-cmd'... ok.
Checking 'tar'... updated.
Checking 'find'... updated.
Checking 'bash'... updated.
Checking 'xargs'... updated.
Checking 'patch'... updated.
Checking 'diff'... updated.
Checking 'cp'... updated.
Checking 'seq'... updated.
Checking 'awk'... updated.
Checking 'grep'... updated.
Checking 'egrep'... updated.
Checking 'getopt'... updated.
Checking 'realpath'... updated.
Checking 'stat'... updated.
Checking 'gzip'... updated.
Checking 'unzip'... updated.
Checking 'bzip2'... updated.
Checking 'wget'... updated.
Checking 'install'... updated.
Checking 'perl'... updated.
Checking 'python'... updated.
Checking 'python3'... updated.
Checking 'python3-distutils'... ok.
Checking 'python3-stdlib'... ok.
Checking 'file'... updated.
Checking 'which'... updated.
Checking 'argp.h'... ok.
Checking 'fts.h'... ok.
Checking 'obstack.h'... ok.
Checking 'libintl.h'... ok.
Checking 'ldconfig-stub'... ok.
Collecting package info: done
Collecting target info: done
Killed
Failed to build tmp/.config-package.in
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a build dependency on 'libpam', which does not exist
WARNING: Makefile 'package/boot/kexec-tools/Makefile' has a dependency on 'liblzma', which does not exist
WARNING: Makefile 'package/network/services/lldpd/Makefile' has a dependency on 'libnetsnmp', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a build dependency on 'libpam', which does not exist
target/Config.in:1: glob failed: No files found "tmp/.config-target.in"
make: *** [/home/catsimple/testingx86/openwrt/include/toplevel.mk:137：menuconfig] 错误 1
```

catsimple commented:
> @catsimple did you ~undo~ revert [74117bb](https://github.com/openwrt/openwrt/commit/74117bbc2fa169c22ce4901d5f0c15235a0e828e)?

I didn't and it works fine after reverting https://github.com/openwrt/openwrt/commit/74117bbc2fa169c22ce4901d5f0c15235a0e828e

vincejv commented on package/kernel/linux/modules/netdevices.mk:
```diff
@@ -1725,7 +1725,7 @@ $(eval $(call KernelPackage,atlantic))
   SUBMENU:=$(NETWORK_DEVICES_MENU)
   TITLE:=Microchip LAN743x PCI Express Gigabit Ethernet NIC
-  DEPENDS:=@PCI_SUPPORT +kmod-ptp +kmod-mdio-devres
+  DEPENDS:=@PCI_SUPPORT +kmod-ptp +kmod-mdio-devres +LINUX_6_6:fixed_phy
```
is it not `kmod-fixed-phy`??

namiltd replied:
In the meantime, I noticed this too.

vincejv commented:
@namiltd Please modify the following lines to try to fix kmod-vfio errors
https://github.com/namiltd/openwrt/blob/86e72fa830051345b04a67435e19f5f8d62fbbe2/package/kernel/linux/modules/virt.mk#L77-L90

I raised this issue also on the previous PR https://github.com/openwrt/openwrt/pull/14751#issuecomment-1977238618 

> As of kernel 6.2+, the vfio_virqfd functionality has been folded into the base vfio module.
> https://github.com/torvalds/linux/commit/e2d55709398e62cf53e5c7df3758ae52cc62d63a

`package/kernel/linux/modules/virt.mk`
```makefile
define KernelPackage/vfio
  SUBMENU:=Virtualization
  TITLE:=VFIO Non-Privileged userspace driver framework
  DEPENDS:=@TARGET_x86_64||TARGET_armsr_armv8
  KCONFIG:= \
	CONFIG_VFIO \
	CONFIG_VFIO_NOIOMMU=n \
	CONFIG_VFIO_MDEV=n
  FILES:= \
	$(LINUX_DIR)/drivers/vfio/vfio.ko \
	$(LINUX_DIR)/drivers/vfio/vfio_virqfd.ko@lt6.2 \
	$(LINUX_DIR)/drivers/vfio/vfio_iommu_type1.ko
  AUTOLOAD:=$(call AutoProbe,vfio vfio_iommu_type1 +LINUX_6_1:vfio_virqfd)
endef
```

Please test on your local first with `make defconfig` or `menuconfig` to see if adjustments are needed to the code below as I'm also not sure. Thank you.

trippleflux commented:
Compiled x64 fine but somehow my x86_64 booting activities only happened at the top left corner of my HDMI monitor after screen firstly enlarged at size of my screen monitor.
![booting-weird](https://github.com/openwrt/openwrt/assets/1216215/0925d3dd-cdb7-4ad8-9085-611264eb4a04)

I have tried this kernel config but still getting the same problem : 
```
CONFIG_DRM_NOMODESET=y
```

Does anyone have a clue? , functionalities wise OpenWrt has no problem only problem mentioned above.

mcbridematt committed:
kernel: vfio: flag vfio_virqfd as only kernel 6.2 and earlier

Kernel 6.2 folded virqfd (eventd interface for VFIO interrupts)
into the base vfio module, it is no longer a tristate option.

Change suggested by vincejv on GitHub:
https://github.com/openwrt/openwrt/pull/14868#issuecomment-1998260124

Signed-off-by: Mathew McBride <matt@traverse.com.au>

vincejv commented:
@trippleflux try adding on your video.mk and include `kmod-drm-i915` in `.config`, but don't think this will get mainlined (this is from coolsnowwolf fork)

```makefile
define KernelPackage/drm-i915
  SUBMENU:=$(VIDEO_MENU)
  TITLE:=Intel GPU drm support
  DEPENDS:=@TARGET_x86 +kmod-drm-buddy +kmod-drm-ttm +kmod-drm-kms-helper +i915-firmware \
	+(LINUX_6_1||LINUX_6_6):kmod-drm-display-helper +(LINUX_6_1||LINUX_6_6):kmod-acpi-video
  KCONFIG:= \
	CONFIG_INTEL_GTT \
	CONFIG_DRM_I915 \
	CONFIG_DRM_I915_CAPTURE_ERROR \
	CONFIG_DRM_I915_COMPRESS_ERROR \
	CONFIG_DRM_I915_DEBUG=n \
	CONFIG_DRM_I915_DEBUG_GUC=n \
	CONFIG_DRM_I915_DEBUG_MMIO=n \
	CONFIG_DRM_I915_DEBUG_RUNTIME_PM=n \
	CONFIG_DRM_I915_DEBUG_VBLANK_EVADE=n \
	CONFIG_DRM_I915_GVT=y \
	CONFIG_DRM_I915_LOW_LEVEL_TRACEPOINTS=n \
	CONFIG_DRM_I915_SELFTEST=n \
	CONFIG_DRM_I915_SW_FENCE_CHECK_DAG=n \
	CONFIG_DRM_I915_SW_FENCE_DEBUG_OBJECTS=n \
	CONFIG_DRM_I915_USERPTR=y \
	CONFIG_DRM_I915_WERROR=n
  FILES:= \
      $(LINUX_DIR)/drivers/gpu/drm/i915/i915.ko
  AUTOLOAD:=$(call AutoProbe,i915)
endef

define KernelPackage/drm-i915/description
  Direct Rendering Manager (DRM) support for "Intel Graphics
  Media Accelerator" or "HD Graphics" integrated graphics,
  including 830M, 845G, 852GM, 855GM, 865G, 915G, 945G, 965G,
  G35, G41, G43, G45 chipsets and Celeron, Pentium, Core i3,
  Core i5, Core i7 as well as Atom CPUs with integrated graphics.
endef

$(eval $(call KernelPackage,drm-i915))
```

trippleflux commented:
@vincejv 
Thank you, recompiling, also trying with this kernel config which was used from woking kernel 6.1 config : 
```
CONFIG_DRM_BUDDY=y
CONFIG_DRM_DISPLAY_DP_HELPER=y
CONFIG_DRM_DISPLAY_HDCP_HELPER=y
CONFIG_DRM_DISPLAY_HDMI_HELPER=y
CONFIG_DRM_DISPLAY_HELPER=y
```

Beginner-Go commented:
![image](https://github.com/openwrt/openwrt/assets/70857188/c600d928-ac8a-4d28-9cfc-dae33585beec)

@namiltd  It seems that `# CONFIG_PROBE_EVENTS_BTF_ARGS is not set` needs to be added to `config-6.8` ?

namiltd commented:
@Beginner-Go What options did you set in menuconfig?

Beginner-Go commented:
> What options did you set in menuconfig?

Sorry I made a mistake

namiltd commented:
@trippleflux Use: https://github.com/openwrt/openwrt/pull/14899

ehem reviewed (changes requested):
There are two scripts available for this, those should be used instead.  Simply copying the files destroys their history making future development troublesome.

One trick with this, the output of those scripts is fragile and cannot be rebased, so this needs special handling.

ehem reviewed (changes requested):
The should should instead be cherry-picked from #14280, as someone was already trying to do this months ago.  Ignoring work done by others is an excellent way to annoy people https://github.com/openwrt/openwrt/pull/14280/commits/6dc0b6ccfa5094e6890f3a8ae0add52f372c1516 (though #14280 is really better for mailing list since quite a few bits can be readily reordered).

namiltd committed:
kernel/x86: Create kernel patches for v6.6 (from v6.1)

This is an automatically generated commit.

During a `git bisect` session, `git bisect --skip` is recommended.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/x86: Restore kernel patches for v6.1

This is an automatically generated commit which aids following Kernel patch history,
as git will see the move and copy as a rename thus defeating the purpose.

See: https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html
for the original discussion.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd reviewed (commented):
[ehem](https://github.com/ehem) Done

graysky2 commented:
Just a heads-up:  I created https://github.com/openwrt/openwrt/pull/14909 which bring 6.6 to 6.6.22 and in doing so will require you edit `target/linux/x86/config-6.6` to add a symbol.  See: https://github.com/openwrt/openwrt/pull/14908/commits/10fd48e77935c82d3ca5e933b0388dea0fb68f72

graysky2 commented:
```
Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne
```

namiltd commented:
@xiaoshengshuohua  I think everything is done and ready to merge.

graysky2 commented:
@namiltd  - do you mind adding two lines to `target/linux/x86/64/config-6.6` like in https://github.com/openwrt/openwrt/pull/13819/files to enable PSI?

namiltd commented:
@graysky2 I think there is no point in duplicating patches. Let's wait until this PR is approved and then yours.

Beginner-Go commented:
Looking forward to the early merger 😊

ehem commented:
Though #14944 has an interaction with this.

trippleflux commented:
@namiltd 
How about mediatek's patches backport?

namiltd commented:
@trippleflux Paste the log fragment containing these errors here or indicate the branch on which you are compiling and the applied corrections along with the config. I will try to recreate your compilation and look at the errors.

trippleflux commented:
@namiltd 
```
error: 'struct mtk_wed_device' has no member named 'rev_id'
```

mt76 fork repo : https://github.com/rany2/openwrt/tree/main/package/kernel/mt76

```
CONFIG_PACKAGE_kmod-mt7915-firmware=y
CONFIG_PACKAGE_kmod-mt7915e=y
CONFIG_PACKAGE_kmod-mt7916-firmware=y
CONFIG_PACKAGE_CFG80211_TESTMODE is not set
```

robimarko reviewed (commented):
Change the title, core_package is definitively not the right one, something like `kernel: modules: netdevices: lan743x: add missing dependency` is better

robimarko reviewed (commented):
Ok, adding comments on patches doesnt seem to work.

robimarko commented on target/linux/generic/hack-6.6/212-tools_portability.patch:
Why is this suddenly not required anymore?

I am also not a fan of this change being included in the x86 kernel bump since it affects all targets

namiltd replied:
On the current version of packages and kernel this removal became unnecessary. I'm just waiting for Github checks to confirm this.

namiltd commented:
[trippleflux](https://github.com/trippleflux) Try copy file 729-09-v6.2-net-ethernet-mtk_wed-add-wcid-overwritten-support-fo.patch from folder target/linux/generic/backport-6.1 to backport-6.6

trippleflux commented:
> [trippleflux](https://github.com/trippleflux) Try copy file 729-09-v6.2-net-ethernet-mtk_wed-add-wcid-overwritten-support-fo.patch from folder target/linux/generic/backport-6.1 to backport-6.6

```
Applying /home/user/works/openwrt/target/linux/generic/backport-6.6/729-09-v6.2-net-ethernet-mtk_wed-add-wcid-overwritten-support-fo.patch using plaintext: 
patching file drivers/net/ethernet/mediatek/mtk_wed.c
Hunk #1 FAILED at 526.
Hunk #2 FAILED at 1290.
Hunk #3 FAILED at 1358.
3 out of 3 hunks FAILED -- saving rejects to file drivers/net/ethernet/mediatek/mtk_wed.c.rej
patching file drivers/net/ethernet/mediatek/mtk_wed_regs.h
Hunk #1 succeeded at 22 with fuzz 2 (offset 2 lines).
patching file include/linux/soc/mediatek/mtk_wed.h
Hunk #1 succeeded at 89 with fuzz 2 (offset 4 lines).
Patch failed!  Please fix /home/user/works/openwrt/target/linux/generic/backport-6.6/729-09-v6.2-net-ethernet-mtk_wed-add-wcid-overwritten-support-fo.patch!
```

Perhaps this patchset 729 is need a rework and couldn't applied partially.

namiltd commented:
Edit patch leaving only:
```
--- a/drivers/net/ethernet/mediatek/mtk_wed_regs.h
+++ b/drivers/net/ethernet/mediatek/mtk_wed_regs.h
@@ -20,6 +20,8 @@ struct mtk_wdma_desc {
 	__le32 info;
 } __packed __aligned(4);
 
+#define MTK_WED_REV_ID					0x004
+
 #define MTK_WED_RESET					0x008
 #define MTK_WED_RESET_TX_BM				BIT(0)
 #define MTK_WED_RESET_TX_FREE_AGENT			BIT(4)
--- a/include/linux/soc/mediatek/mtk_wed.h
+++ b/include/linux/soc/mediatek/mtk_wed.h
@@ -85,6 +85,9 @@ struct mtk_wed_device {
 	int irq;
 	u8 version;
 
+	/* used by wlan driver */
+	u32 rev_id;
+
 	struct mtk_wed_ring tx_ring[MTK_WED_TX_QUEUES];
 	struct mtk_wed_ring rx_ring[MTK_WED_RX_QUEUES];
 	struct mtk_wed_ring txfree_ring;

trippleflux commented:
> --- a/drivers/net/ethernet/mediatek/mtk_wed_regs.h
> +++ b/drivers/net/ethernet/mediatek/mtk_wed_regs.h
> @@ -20,6 +20,8 @@ struct mtk_wdma_desc {
>  	__le32 info;
>  } __packed __aligned(4);
>  
> +#define MTK_WED_REV_ID					0x004
> +
>  #define MTK_WED_RESET					0x008
>  #define MTK_WED_RESET_TX_BM				BIT(0)
>  #define MTK_WED_RESET_TX_FREE_AGENT			BIT(4)
> --- a/include/linux/soc/mediatek/mtk_wed.h
> +++ b/include/linux/soc/mediatek/mtk_wed.h
> @@ -85,6 +85,9 @@ struct mtk_wed_device {
>  	int irq;
>  	u8 version;
>  
> +	/* used by wlan driver */
> +	u32 rev_id;
> +
>  	struct mtk_wed_ring tx_ring[MTK_WED_TX_QUEUES];
>  	struct mtk_wed_ring rx_ring[MTK_WED_RX_QUEUES];
>  	struct mtk_wed_ring txfree_ring;

Thanks but if you look at the 729-09 patch I couldn't count on that partial patch as it might needs others patches in that patchset 729.

I am just afraid that it might affect mt76 functionalities wise. Don't mind me, i will just wait for more proper patches.

namiltd committed:
generic: 6.6: Removal of tools_portability.patch already included in kernel 6.6

Removal of tools_portability.patch already included in kernel 6.6

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/x86: Create kernel files for v6.6 (from v6.1)

This is an automatically generated commit.

During a `git bisect` session, `git bisect --skip` is recommended.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
kernel/x86: Restore kernel files for v6.1

This is an automatically generated commit which aids following Kernel patch history,
as git will see the move and copy as a rename thus defeating the purpose.

See: https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html
for the original discussion.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

the2masters commented on target/linux/x86/config-6.6:
```diff
@@ -212,6 +218,8 @@ CONFIG_KEXEC=y
 CONFIG_KEYBOARD_ATKBD=y
 CONFIG_KMAP_LOCAL=y
+# CONFIG_KVM_PROVE_MMU is not set
+# CONFIG_KVM_SMM is not set
```
I'm running a qemu/kvm on an x86 openwrt host, and this breaks it.
`qemu-system-x86_64: System Management Mode not supported by this hypervisor.`

Please remove this line and add it to package/kernel/linux/modules/virt.mk (see my other comment)

namiltd replied:
patch removed

the2masters commented on package/kernel/linux/modules/virt.mk:
and add it here in line 25. Then qemu boots the virtual machines again.
```
diff --git a/package/kernel/linux/modules/virt.mk b/package/kernel/linux/modules/virt.mk
index a408ac9b14..d0aa63f195 100644
--- a/package/kernel/linux/modules/virt.mk
+++ b/package/kernel/linux/modules/virt.mk
@@ -22,6 +22,7 @@ define KernelPackage/kvm-x86
   KCONFIG:=\
          CONFIG_KVM \
          CONFIG_KVM_MMU_AUDIT=n \
+         CONFIG_KVM_SMM=y \
          CONFIG_VIRTUALIZATION=y
   FILES:= $(LINUX_DIR)/arch/$(LINUX_KARCH)/kvm/kvm.ko
   AUTOLOAD:=$(call AutoProbe,kvm.ko)
```

the2masters commented:
Thank you for your instant reply and changes!

Tested-by: Stefan Hellermann <stefan@the2masters.de>
runs on a x86-64 intel laptop in efi mode and on two x86-64 virtual machines without efi in qemu-kvm on this host.

ehem commented on target/linux/x86/generic/config-6.6:
```diff
@@ -31,7 +32,7 @@ CONFIG_ACPI_TAD=y
 # CONFIG_ACPI_TOSHIBA is not set
 CONFIG_ACPI_VIDEO=y
-# CONFIG_ACPI_WMI is not set
+CONFIG_ACPI_WMI=y
```
Seeing this a bunch.  This doesn't seem appropriate for a _refresh_.  I can believe it is necessary, just doesn't seem right for this commit.

namiltd replied:
fixed

ehem commented on target/linux/x86/generic/config-6.6:
```diff
@@ -44,36 +45,43 @@ CONFIG_AGP_INTEL=y
 CONFIG_CPU_FREQ_DEFAULT_GOV_SCHEDUTIL=y
 CONFIG_CPU_FREQ_GOV_SCHEDUTIL=y
-# CONFIG_CPU_IDLE_GOV_HALTPOLL is not set
+CONFIG_CPU_IDLE_GOV_HALTPOLL=y
```
Another one being flipped.

namiltd replied:
fixed

ehem commented on target/linux/x86/geode/config-6.6:
```diff
@@ -35,43 +35,36 @@ CONFIG_ACPI_THERMAL=y
 # CONFIG_AMD_PMF is not set
-CONFIG_ARCH_MIGHT_HAVE_ACPI_PDC=y
 # CONFIG_ASUS_TF103C_DOCK is not set
+CONFIG_ARCH_MIGHT_HAVE_ACPI_PDC=y
```
I agree using ASCII ordering is better.  Alas this one the order became wrong.

namiltd replied:
fixed

ehem commented on target/linux/x86/legacy/config-6.6:
```diff
@@ -28,7 +29,7 @@ CONFIG_ACPI_SYSTEM_POWER_STATES_SUPPORT=y
 CONFIG_ACPI_THERMAL=y
 # CONFIG_ACPI_TOSHIBA is not set
 CONFIG_ACPI_VIDEO=y
-# CONFIG_ACPI_WMI is not set
+CONFIG_ACPI_WMI=y
```
Again, _refresh_?

ehem reviewed (commented):
Alas, I'll agree with changing option order to match ASCII (alas, '_' is after numbers).  A few of the configuration changes don't really seem appropriate for a _refresh_.  They may be appropriate, but not really _refresh_.

namiltd committed:
x86: 6.6: enable System Management Mode emulation on KVM

Include the CONFIG_KVM_SMM option in the kvm-x86 package to enable system management mode emulation on KVM.

Co-authored-by: Stefan Hellermann <stefan@the2masters.de>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented on target/linux/x86/64/config-6.6:
```diff
@@ -515,6 +559,7 @@ CONFIG_VIRTIO_NET=y
 CONFIG_VIRTIO_PCI=y
 CONFIG_VIRTIO_PCI_LEGACY=y
 CONFIG_VIRTIO_PCI_LIB=y
+CONFIG_VIRTIO_PCI_LIB_LEGACY=y
```
It is new option.

namiltd commented on target/linux/x86/64/config-6.6:
```diff
@@ -74,60 +77,76 @@ CONFIG_BLK_DEV_NVME=y
 CONFIG_CPU_IBPB_ENTRY=y
 CONFIG_CPU_IBRS_ENTRY=y
-# CONFIG_CPU_IDLE_GOV_HALTPOLL is not set
+CONFIG_CPU_IDLE_GOV_HALTPOLL=y
```
fixed

namiltd commented on target/linux/x86/64/config-6.6:
```diff
@@ -37,33 +38,35 @@ CONFIG_ACPI_SYSTEM_POWER_STATES_SUPPORT=y
 CONFIG_ACPI_VIDEO=y
-# CONFIG_ACPI_WMI is not set
+CONFIG_ACPI_VIOT=y
+CONFIG_ACPI_WMI=y
```
fixed

namiltd commented:
@ehem All done.

robimarko commented:
@ehem Any more remarks?

ehem commented on target/linux/x86/geode/config-6.6:
```diff
@@ -35,43 +35,36 @@ CONFIG_ACPI_THERMAL=y
 # CONFIG_CS89x0_ISA is not set
 CONFIG_DMA_ACPI=y
 # CONFIG_EL3 is not set
+CONFIG_FUNCTION_ALIGNMENT=16
+CONFIG_FUNCTION_ALIGNMENT_16B=y
+CONFIG_FUNCTION_PADDING_BYTES=16
+CONFIG_FUNCTION_PADDING_CFI=11
```
This surprises me.  I could believe Geode likes 16-byte aligned functions, but as a late-model 32-bit platform I would expect 4-byte or 8-byte alignment.  I don't have the 6.6 source in front of me nor I am particularly familiar with the Geode processors.

ehem reviewed (changes requested):
Alas, I did find some things to comment on.  Sorry about being a bit slow to update.

namiltd commented on package/kernel/linux/modules/virt.mk:
```diff
@@ -84,9 +84,9 @@ define KernelPackage/vfio
   FILES:= \
 	$(LINUX_DIR)/drivers/vfio/vfio.ko \
-	$(LINUX_DIR)/drivers/vfio/vfio_virqfd.ko \
+	$(LINUX_DIR)/drivers/vfio/vfio_virqfd.ko@lt6.2 \
```
@lt6.2 means less than or equal to 6.1
It could be replaced with @le6.1 but it doesn't change anything

robimarko replied:
Dont bother with it

namiltd committed:
kernel: 6.6: modules: netdevices: lan743x: add missing dependency

Fix error: Package kmod-lan743x is missing dependencies for the following libraries:
fixed_phy.ko

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

robimarko commented on package/kernel/linux/modules/netdevices.mk:
```diff
@@ -1742,7 +1742,7 @@ $(eval $(call KernelPackage,atlantic))
   SUBMENU:=$(NETWORK_DEVICES_MENU)
   TITLE:=Microchip LAN743x PCI Express Gigabit Ethernet NIC
-  DEPENDS:=@PCI_SUPPORT +kmod-ptp +kmod-mdio-devres
+  DEPENDS:=@PCI_SUPPORT +kmod-ptp +kmod-mdio-devres +LINUX_6_6:kmod-fixed-phy
```
No, this doesn't work as 5.15 doesn't need it, it will require manual fixups regardless.

robimarko commented:
After CI is done I plan to merge this

namiltd committed:
x86: 6.6: refresh kernel config

Refresh kernel config.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
@robimarko All done

namiltd commented:
Is there a chance for approval before the kernel is updated to 6.6.24 (released today) and we have to adapt the patches again?

graysky2 commented:
@namiltd - https://github.com/openwrt/openwrt/pull/15063

ehem reviewed (approved):
This looks good enough to me.  Ultimately my real issue is I dislike the way kernel configurations are handled in OpenWRT.  As such seems best to get this in now.

robimarko commented:
Thanks! Rebased on top of main and merged!
